aiosysbus/access.py

```diff
@@ -69,40 +79,45 @@ def _get_session_token(self):
             except RequestException as error:
                 raise HttpRequestError("Error HttpRequest") from error
 
-            resp = r.json()
+            resp: dict[str, dict[str, Any]] = r.json()
             if not resp.get("data"):
                 raise AuthorizationError(
                     "Starting session failed (APIResponse: {})".format(json.dumps(resp))
                 )
 
-            session_token = resp.get("data").get("contextID")
-            session_permissions = resp.get("data").get("groups")
+            session_token: str | None = resp["data"].get("contextID")
+            session_permissions: str | None = resp["data"].get("groups")
             logger.debug("Token %s", session_token)
 
             return (session_token, session_permissions)
+        return (None, None)
 
-    def _refresh_session_token(self):
+    def _refresh_session_token(self) -> None:
         """Refresh token."""
         # Get token for the current session
         session_token, session_permissions = self._get_session_token()
         self.session_token = session_token
         self.session_permissions = session_permissions
 
-    def _retry(self, response, callback, *kwargs):
+    def _retry(
+        self, response: Response, callback: Callable[..., Any], *kwargs: Any
+    ) -> None:
         if response.status_code == 401 and self.retry < self.max_retry:
             self.retry += 1
             callback(kwargs)
         else:
             self.retry = 0
 
-    def _get_headers(self):
+    def _get_headers(self) -> dict[str, str]:
         """Get headers."""
         return {
-            "X-Context": self.session_token,
+            "X-Context": self.session_token,  # type: ignore[dict-item]
```

Comment:
Maybe it should raise if session_token is None?
For now I added `type: ignore` to keep existing behavior.

Reply:
I agree, it is necessary to throw an exception